forum_leaderboard/forum_leaderboard.py

```diff
@@ -78,27 +108,75 @@ def student_view(self, context=None):
         return self.create_fragment(
             "static/html/forum_leaderboard.html", context=context,
             css=["static/css/forum_leaderboard.css"],
-            javascript=["static/js/src/forum_leaderboard.js"],
-            initialize='ForumLeaderboardXBlock')
+        )
 
     def author_view(self, context=None):
         return self.create_fragment(
             "static/html/forum_leaderboard_studio.html",
             context={
                 'discussion_id': self.discussion_id,
-                'display_name': self.display_name
+                'display_name': self.display_name,
+                'count': self.count,
             },
-            css=["static/css/forum_leaderboard.css"])
+            css=["static/css/forum_leaderboard.css"]
+        )
+
+    def studio_view(self, context=None):
+        return self.create_fragment(
+            "static/html/forum_leaderboard_studio_edit.html",
+            context={'discussion_id': self.discussion_id, 'count': self.count},
+            css=["static/css/forum_leaderboard.css"],
+            javascript=["static/js/src/forum_leaderboard_studio.js"],
+            initialize='ForumLeaderboardStudioXBlock'
+        )
+
+    @XBlock.json_handler
+    def studio_submit(self, data, suffix=''):
+        result = {'success': True, 'errors': []}
+        try:
+            count = int(data.get('count', ForumLeaderboardXBlock.count.default))
+            if not count > 0:
```

Comment:
That could also work, but I felt this was more semantic to how I said it out loud in my head. I can change it if it's bothersome, though.